docs/user-guide/pods/init-container.md

```diff
@@ -0,0 +1,168 @@
+---
+assignees:
+- erictune
+
+---
+
+* TOC
+{:toc}
+
+In addition to having one or more main containers (or **app containers**), a
+pod can also have one or more **init containers** which run before the app
+containers.   Init containers allow you to reduce and reorganize setup scripts
+and "glue code".
+
+## Summary
+
+An init container is exactly like a regular container, except that it always
+runs to completion and each init container must complete successfully before
+the next one is started. If the init container fails (exits with a non-zero
+exit code) on a `RestartNever` pod the pod will fail - otherwise it will be
+restarted until it succeeds or the user deletes the pod.
+
+Init containers are currently specified by adding an annotation.  The
+annotation key is `pod.beta.kubernetes.io/init-containers`.  The annotation
+value is a JSON array of [objects of type `v1.Container`
+](http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_container)
+
+Once the feature exits beta, the init containers will be specified on the Pod
+Spec alongside the app `containers` array.
+The status of the init containers is returned as another annotation -
+`pod.beta.kubernetes.io/init-container-statuses` -- as an array of the
+container statuses (similar to the `status.containerStatuses` field).
+
+Init containers support all of the same features as normal containers,
+including resource limits, volumes, and security settings. The resource
+requests and limits for an init container are [handled slightly differently](
+#resources).  Init containers do not support readiness probes since they will
+run to completion before the pod can be ready.
+
+An init container has all of the fields of an app container.  A pod can have
+zero or more init containers.  The init containers run one at a time in order.
+Each much succeed before the next can run.  Once that happens, the pod is
+initialized, and the app containers run as usual.
+
+## Use Cases
+
+Because init containers have separate images from app containers:
+
+* they can contain utilities that are not desirable to include in the app container
+  image for security reasons,
+* they can contain utilities or custom code for setup that is not present in an app
+  image.  (No need to make an image `FROM` another image just to use a tool like
+  `sed`, `awk`, `python`, `dig`, etc during setup).
+* the application image builder and the deployer roles can work independently without
+  the need to jointly build a single app image.
+
+Because init containers have different filesystem view (Linux namespaces) from
```

Comment:
No point in a bullet list with one bullet. Just make it all into one sentence.

"In addition, because init containers have different filesystem views (Linux namespaces) from app containers, they can be given access to Secrets that the app containers are not able to access."